.github/workflows/weekly_update.yml

```
@@ -0,0 +1,39 @@
+name: Autoupdate
+
+on:
+  schedule:
+    - cron: "0 0 * * 1"  # every Monday at midnight
+
+jobs:
+  weekly_update:
+    name: Weekly update
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Configure Git
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+      - name: Create new branch
+        run: |
+          git checkout -b weekly_update
```

Checkout an existing branch, if any:
```suggestion
          git checkout weekly_update || git checkout -b weekly_update
```